packages/rest/src/__tests__/unit/router/rebase-openapi-spec.unit.ts

```diff
@@ -0,0 +1,143 @@
+    };
+    const rebasedSpec = rebaseOpenApiSpec(spec, '/api');
+
+    expect(spec).to.eql(rebasedSpec);
+  });
+
+  it('does not modify the OpenApiSpec if basePath is empty or `/`', async () => {
+    const spec = {
+      paths: {
+        '/': {
+          get: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/hello': {
+          post: {
+            summary: 'says hello',
+            consumes: 'application/json',
+            responses: {
+              '200': {
+                description: 'OK',
+              },
+            },
+          },
+        },
+      },
+    };
+
+    let rebasedSpec = rebaseOpenApiSpec(spec, '');
+    expect(spec).to.eql(rebasedSpec);
+
+    rebasedSpec = rebaseOpenApiSpec(spec, '/');
+    expect(spec).to.eql(rebasedSpec);
+  });
+
+  it('rebases OpenApiSpec if there is a basePath', async () => {
+    const spec = {
+      paths: {
+        '/': {
+          get: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/hello': {
+          post: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const rebasedSpec = rebaseOpenApiSpec(spec, '/greetings');
+    const rebasedPaths = Object.keys(rebasedSpec.paths);
+
+    expect(rebasedPaths).to.eql(['/greetings/', '/greetings/hello']);
+  });
+
+  it('does not modify the original OpenApiSpec', async () => {
+    const spec = {
+      paths: {
+        '/': {
+          get: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+        '/hello': {
+          post: {
+            responses: {
+              '200': {
+                description: 'greeting',
+                content: {
+                  'application/json': {
+                    schema: {type: 'string'},
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    rebaseOpenApiSpec(spec, '/greetings');
+
+    expect(spec).to.equal(spec);
```

Comment:
This is a [Tautology](https://en.wikipedia.org/wiki/Tautology_(logic)) - you are asserting that `spec` is a pointer to the same object as pointed to by `spec`.

I think you should check that `spec.paths` has entries `/` and `/hello` after the rebase.

```suggestion
    expect(Object.keys(spec.paths)).to.deepEqual('/', '/hello');
```